pandas/core/arrays/boolean.py

```diff
@@ -130,9 +130,19 @@ def coerce_to_array(values, mask=None, copy: bool = False):
     if isinstance(values, np.ndarray) and values.dtype == np.bool_:
         if copy:
             values = values.copy()
+    elif isinstance(values, np.ndarray) and values.dtype in (np.int_, np.float_):
```

Comment:
I think this check is incorrect.
```python
a = np.array([1], dtype=np.int8)
a.dtype == np.int_   # False
```

Should this be using the dtype checking functions from pandas.dtypes.common (like `is_numeric_dtype`)?